lmfdb/elliptic_curves/templates/ec-curve.html

```diff
@@ -467,7 +467,8 @@ <h2> {{KNOWL('ec.galois_rep', title='Galois representations')}} </h2>
 {% endif %}
 
 {% if data.adelic_level %}
-<p> The image of the {{KNOWL('ec.galois_rep', title='adelic Galois representation')}} has {{KNOWL('ec.galois_rep_adelic_image', title='level')}} ${{data.adelic_level}}$, {{KNOWL('ec.galois_rep_adelic_image', title='index')}} ${{data.adelic_index}}$, and {{KNOWL('ec.galois_rep_adelic_image', title='genus')}} ${{data.adelic_genus}}$.</p>
+<p> The image of the {{KNOWL('ec.galois_rep', title='adelic Galois representation')}} has {{KNOWL('ec.galois_rep_adelic_image', title='level')}} ${{data.adelic_level}}$, {{KNOWL('ec.galois_rep_adelic_image', title='index')}} ${{data.adelic_index}}$, {{KNOWL('ec.galois_rep_adelic_image', title='genus')}} ${{data.adelic_genus}}$, and generators</p>
+<p>${{data.data.adelic_data.adelic_gens}}$</p>
```

Comment:
It looks like there's an extra data here?  If you're only using the gens, why not just change the code so that you can write this is `data.adelic_gens`?

Reply:
This extra `data` is used in many other places in this file, eg:

https://github.com/LMFDB/lmfdb/blob/bbadf384c6306d82de37fd7b13b3ee2094cd0106/lmfdb/elliptic_curves/templates/ec-curve.html#L445

It seems to arise when a database lookup is done from another table. In the above line, `data.adelic_level` works because `adelic_level` is a column in the `ec_curvedata` table, but when we add other things in the data blob in `web_ec.py`, it becomes `data.data`. I wouldn't know how to get around this ...

Reply:
This is a known issue with how web_ec is implemented, changing it would require a lot of changes that are outside the scope of this PR.  I'm not sure it is worth doing, but if it is, I think it should be done in a separate PR.